### PR TITLE
Add gcw0 platform.

### PR DESCRIPTION
**Pull request by john-parton**
Changes to get dosbox-pure to compile with the rg350/opendingux toolchain. Resolves https://github.com/schellingb/dosbox-pure/issues/79

You have to put the toolchain at `/opt/gcw0-toolchain/` and then you can cross-compile by running `make platform=gcw0 -j8`.

Install by copying `dosbox_pure_libretro.so` to `~/.retroarch/cores/` (SSH or eject SD card)

Tested with latest RetroArch nightly built by the build bot. 

I *did* enable `DYNAREC`, but I think there might be more work to do here. I noticed the dynarec code on this repo has some differences:  https://github.com/dvk11/dosbox-rg350/blob/master/src/cpu/core_dynrec/risc_mipsel32.h

Performance isn't great. The biggest issue is the sound is VERY crackle-y and distorted. I can try to take a video of performance.

I don't have a keyboard on my test device at this time. Will probably invest in one soon.

### CODE REVIEW

**Comment by schellingb** [comment]
Can you tell me what errors you run into when compiling with `C_DBP_SUPPORT_MIDI_TSF` enabled?
I'm the author of TSF (TinySoundFont) so I'd love to make it work :-)

File: src/dos/cdrom_image.cpp [hunk 1]
```diff
@@ -534,7 +534,7 @@ bool CDROM_Interface_Image::LoadCueSheet(char *cuefile)
 	}
 	df->Close();
 	delete df;
-	istringstream inString = istringstream(dosfilebuf);
+	istringstream inString; //  = istringstream(dosfilebuf);
```
Comment (john-parton) [inline src/dos/cdrom_image.cpp:537]: This is probably bad. What's this even doing?

**Comment by john-parton** [comment]
# Update

Compiles & Runs!

Performance on the RG350M is pretty terrible. Will try adjusting some flags.

This change right here: https://github.com/schellingb/dosbox-pure/pull/80/files#diff-03e5d4c65c7133961ad0e9600959bf4a9d74ae3d51bc52fb28f8a97e2f8c7b38R537

Probably breaks things. Not sure what that even does.

**Comment by john-parton** [comment]
I experimented with some flags. I think I got the sweet spot for performance for now.

This is the standalone dosbox emulator that I have been comparing it to: https://github.com/dvk11/dosbox-rg350

A few games run much slower: jill.exe and oregon trail deluxe are two that I found

Might be because that branched off from dosbox 8 years ago and there was a performance regression, or it could be some issue with the libretro port, or dosbox-pure specifically. I don't know.

I found one game which runs fine on the old fork, but crashes in the current one: Jazz Jackrabbit.

**Comment by john-parton** [comment]
Here's a sheet with some games that I tested for compatibility: https://docs.google.com/spreadsheets/d/1PMpXXqJZ6gWxvUgmIIOeUVYuw9lVIPIBdvyP5QKh7q0/edit#gid=469661062

**Comment by twinaphex** [comment]
Hi @schellingb , is this fit for merge?

File: src/dos/cdrom_image.cpp (hunk 1 above)
Comment (schellingb) [reply src/dos/cdrom_image.cpp:537]: This needs to be `istringstream inString(dosfilebuf);`.

**Comment by schellingb** [comment]
@twinaphex Good question, it looks good except one thing:

@john-parton I replied to your review of src/dos/cdrom_image.cpp. Can you fix that? Then I think I can merge this.

File: src/dos/cdrom_image.cpp (hunk 1 above)
Comment (john-parton) [reply src/dos/cdrom_image.cpp:537]: I'll get this fixed today

**Comment by john-parton** [comment]
I'll get this updated today

**Comment by john-parton** [comment]
Pushed that fix. Let me know if you need additional changes

**Pull request merged**